tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,11 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return self.__repr_()
+
+  def __repr__(self):
+    return "<tf.Variable '%s' shape=%s dtype=%s>" % (
+            self.name(), self.get_shape(), self.dtype.name)
```

Comment:
Done! Sorry about the delay; bazel run super slow when your other computer isn't a datacenter :P

One quick thing about the test: I couldn't figure out why the dtype name has a `_ref` postfix when testing